### PR TITLE
DOC: clarify residual in np.polyfit

### LINKED ISSUE #7213: polyfit reference: definition of 'residuals'
Returns:  
residuals, rank, singular_values, rcond :
Present only if full = True. Residuals of the least-squares fit...

The 'residuals' returned is  the sum of (the square of each residual) (i.e. one number in an array).  It would be useful to actually return the array of all the residuals; if not that, then documentation should be changed.

### PR DESCRIPTION
**Pull request by sethtroisi**
Closes #7213

wording matches `resid -- sum of squared residuals of the least squares fit` which is present in several files ([polynomial.py](https://github.com/numpy/numpy/blob/master/numpy/polynomial/polynomial.py#L1251), _polybase.py, hermite.py, ...)

### CODE REVIEW

File: numpy/lib/polynomial.py [hunk 1]
```diff
@@ -479,10 +479,10 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
...
 
     residuals, rank, singular_values, rcond
-        Present only if `full` = True.  Residuals of the least-squares fit,
-        the effective rank of the scaled Vandermonde coefficient matrix,
-        its singular values, and the specified value of `rcond`. For more
-        details, see `linalg.lstsq`.
+        Present only if `full` = True.  Residuals is sum of squared residuals
+        of the least-squares fit, the effective rank of the scaled Vandermonde
+        coefficient matrix, its singular values, and the specified value of
+        `rcond`. For more details, see `linalg.lstsq`.
```
Comment (bsipocz) [inline numpy/lib/polynomial.py]: @mattip - these attempted links to parameters should raise sphinx warnings, and ideally the docs build test should fail on those?

Comment (mattip) [reply numpy/lib/polynomial.py]: In order for sphinx to catch these, we need to run with `-n`, which currently spits out over 1000 warnings. xref gh-14699 which fixes the common and non-controversial ones, and issue gh-14699 for tracking progress

Comment (bsipocz) [reply numpy/lib/polynomial.py]: Oh, right, indeed. I assumed without much thinking that this was turned on already, sorry for the noise then. 
If an extra pair of hand would help, let me know, messing with sphinx is well within my comfort zone.

Comment (mattip) [reply numpy/lib/polynomial.py]: a review of #14699 would be helpful, and continuation of the work would be great

Comment (rgommers) [reply numpy/lib/polynomial.py]: These shouldn't raise warnings, usage is correct here. At least, this is what the numpydoc spec says. And the rendering can be controlled, see https://github.com/numpy/numpydoc/issues/233

**Pull request merged**